### DISCUSSION TITLE
Akku-Limiter-Mode, Neue Idee für die „Berechnung“ der aktuellen minimalen Akkuspannung (Benötigt keinen Korrekturfaktor)

### DISCUSSION THREAD

**Discussion by SW-Niko** [hoylabs/OpenDTU-OnBattery#664] [Ideas]
Durch die regen Diskussionen in den letzten Tagen bin ich auf einen etwas anderen Ansatz gekommen.

Warum:
Das aktuelle Konzept mit dem Korrekturfaktor funktioniert gut bei einfachen Anlagen (1 Inverter, 1 Akku, 1 Laderegler) und bei richtiger Konfiguration des Korrekturfaktors. Aber bei komplizierteren Anlagen (Mehrere Inverter, Akkus, Laderegler) wird die Berechnung immer komplizierter.
Zusätzlich habe ich bedenken das der Korrekturfaktor bei einer Anlage statisch ist. Der Widerstand der Kupferleitungen und Sicherungen ändert sich mit der Temperatur und der Innenwiderstand der Akkuzellen verändert sich vermutlich mit der Temperatur und der der Höhe der Entnahmeleistung.

Akku-Limiter-Mode-Prinzip: 
Anstatt mit der Entnahmeleistung und einem Korrekturfaktor die Spannung in den Akkuzellen zu berechnen reduzieren wir die Entnahmeleistung schrittweise bis die konfigurierte minimale Akkuspannung am verfügbaren Spannungsmesspunkt (Inverter, MPPT, BMS) nicht mehr unterschritten wird. Mit anderen Worten. wir führen im Akkubetrieb eine dynamische Leistungsbegrenzung ein.

Vorteile:
- Ein Korrekturfaktor wird nicht mehr benötigt.
- Bei Anlagen mit hohem Korrekturfaktor (24V, lange Leitungen, kleine Kapazität) wird der Inverter nicht mehr so stark unterhalb der minimalen Akkuspannung betrieben. Thema Strombegrenzung bei den Hoymiles.
- Reduzierung der Leitungs- und Batterieverluste im Akkubetrieb. 
- Die Batterie wird schrittweise, mit sinkender Ladung, sanfter entladen.
- Der Einfluss des dynamischen Innenwiderstandes der Batterie wird reduziert.
- Das könnte auch die Lösung für [Request] DPL: Limit Battery discharge current #657 sein.

Nachteil:
- Im Grenzbereich wenn der Akku bereits im Akku-Limiter-Modus läuft und dann wieder Energie ankommt (Morgens Sonne) könnte die gesamt Energieentnahme kleiner sein als im derzeitigen Voll-Betrieb.

Wie kommt man wieder aus dem Akku-Limiter-Betrieb raus?

Ich sehe 3 Fälle.
1. Der Akku wird bis zur konfigurierten minimal Spannung entladen und der Inverter wird ausgeschaltet.
Dann erfolgt die Einschaltung wie bisher, sobald die obere Einschaltschwelle wieder erreicht wird.
2. Im Akku-Limiter-Betrieb kommt wieder Energie von der Sonne. Dann wird schrittweise das Limit wieder erhöht bis die konfigurierte obere Leistungs-Grenze erreicht wird.
3. Bei Aus- Einschalten der DTU oder Reset bleibt alles wie es ist.

Geschwindigkeit der Regelung:
Wir bleiben bei der aktuellen Regelgeschwindigkeit für den Inverter von ca 5 Sekunden. Das reicht um der Entladung des Akkus zu folgen und bei plötzlich einsetzendem Sonnenschein den Akku-Limiter Modus innerhalb 30-60 Sekunden wieder zu verlassen.

Bitte mal das Konzept prüfen. Gibt es gravierende Nachteile?

**Comment by SW-Niko** [comment]
Hallo @spcqike / @Manos1966,
Habt ihr schon einen Blick auf die Idee geworfen. Wie ist eure Meinung zu der Idee?

**Comment by Manos1966** [comment]
Hall @SW-Niko 

Ich habe zu wenig Ahnung um eine qualifizierte Meinung darueber zu geben....

Das habe ich ich nicht verstanden:
> Anstatt mit der Entnahmeleistung und einem Korrekturfaktor die Spannung in den Akkuzellen zu berechnen,
 > reduzieren wir die Entnahmeleistung schrittweise bis die konfigurierte minimale Akkuspannung am verfügbaren Spannungsmesspunkt (Inverter, MPPT, BMS) nicht mehr unterschritten wird. Mit anderen Worten. wir führen im Akkubetrieb eine dynamische Leistungsbegrenzung ein.

Meine Feststellungen sind,
- die Batterie Spannung verringert sich, je hoeher die Entnahme ist. Der Unterschied ist sogar beachtlich, wenn man grossere Wechselrichter anschliesst z.B. 1500W (fuer Test-Zwecke) .
- die Batterie Zellen haben eine kleine(re) Spannungsdifferenz (verglichen zu Bleibatterien)
- das grossere Problem ist vor allem Vormittags bei wenig Sonne, wenn die Batterie leer ist z.B. 15% und der Victron die Batterie laedt (Spannung ist hoch, ca 53V) aber die Ladeleistung ist kleiner z.B. 500W, als die Maximale Leistung des Wechselrichters z.B. 1500W. 
Dann passiert ein "Tisch-Tennis" Effekt : Spannung 53V (laden), Victron liefert nur 400W (es ist bewoelkt), Wechselrichter startet volle Leistung 1500W = Die Spannung geht runter, der Wechselrichter stoppt. Und es faengt wieder von vorne.... das macht die fast leere Batterie noch leerer.
 
![00 Spannungen der Pylontech US-Serie](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/14abbefc-2c47-4423-906a-f5e0066cfc2e)

**Reply by SW-Niko** [reply]
>das grossere Problem ist vor allem Vormittags bei wenig Sonne, wenn die Batterie leer ist z.B. 15% und der Victron die Batterie laedt (Spannung ist hoch, ca 53V) aber die Ladeleistung ist kleiner z.B. 500W, als die Maximale Leistung des Wechselrichters z.B. 1500W.
Dann passiert ein "Tisch-Tennis" Effekt : Spannung 53V (laden), Victron liefert nur 400W (es ist bewoelkt), Wechselrichter startet volle Leistung 1500W = Die Spannung geht runter, der Wechselrichter stoppt. Und es faengt wieder von vorne.... das macht die fast leere Batterie noch leerer.

Ich sehe gute Chancen, das der Akku-Limiter-Mode auch dieses Problem behebt. Ich melde mich wenn ich einige Berechnungen und Überlegungen durchgeführt habe.

**Reply by SW-Niko** [reply]
Der Load-Correction-Factor-Mode rechnet falsch in einigen Fällen. Wenn die Widerstände in deiner Anlage und/oder die Ströme hoch genug sind und/oder der Load-Correction-Factor zu hoch angesetzt ist dann kann die berechnete Spannung je nach Bewölkung blitzschnell zwischen der Einschaltspannung und der Ausschaltspannung hin und her schwanken und deinen Effekt verursachen.
Das müsste man aber im Logging sehen.

**Comment by spcqike** [comment]
hallo,

ja, ich hab mir deinen Beitrag auch durchgelesen, leider habe ich auch ihn nicht vollständig verstanden, denke ich :) Oder einfach eine etwas andere Sichtweise.

Grundsätzlich ja: Ich denke, eine Anpassung des DPL oder eine Verbesserung des Umgangs mit Batterien im Allgemeinen, ist sinnvoll.

> Zusätzlich habe ich bedenken das der Korrekturfaktor bei einer Anlage statisch ist. Der Widerstand der Kupferleitungen und Sicherungen ändert sich mit der Temperatur und der Innenwiderstand der Akkuzellen verändert sich vermutlich mit der Temperatur und der der Höhe der Entnahmeleistung.

Ja, das passiert definitiv. Aber der Innenwiderstand ändert sich auch im Laufe der Lebenszeit (SoH, State of Health). Ich habe hin und wieder das Gefühl, dass die Leute mit dem Korrekturfaktor auch den Spannungsabfall in der Batterie kompensieren wollen. dem würde ich tunlichst widersprechen. Der Korrekturfaktor sollte nur von den Schraubverbindungen der Batterie bis zu den Verbindungen am Wechselrichter alle auftretenden Leitungswiderstände kompensieren. (ggf. noch die Leistungstransistoren/Elemente im BMS, aber definitiv nicht den Innenwiderstand der Zelle an sich!)

> Akku-Limiter-Mode-Prinzip:
Anstatt mit der Entnahmeleistung und einem Korrekturfaktor die Spannung in den Akkuzellen zu berechnen reduzieren wir die Entnahmeleistung schrittweise bis die konfigurierte minimale Akkuspannung am verfügbaren Spannungsmesspunkt (Inverter, MPPT, BMS) nicht mehr unterschritten wird. Mit anderen Worten. wir führen im Akkubetrieb eine dynamische Leistungsbegrenzung ein.

klingt nicht verkehrt

> Ein Korrekturfaktor wird nicht mehr benötigt.

finde ich nicht richtig, den Korrekturfaktor zum Kompensieren von Spannungsabfällen zwischen Batterie und Inverter, würde ich beibehalten. Je nach verfügbarer Spannungsquelle (BMS, SmartShunt, Laderegler, Invertereingang) wird benötigt, oder auch nicht. (eigentlich braucht man ihn nur beim Invertereingang, ggf. noch beim Laderegler, wenn dieser keine Senseleitung hat)

Ohne Faktor bei den 2 Quellen, führt das zu einem verfrühten Rückgang der möglichen Entnahmeleistung. 

> Bei Anlagen mit hohem Korrekturfaktor (24V, lange Leitungen, kleine Kapazität) wird der Inverter nicht mehr so stark unterhalb der minimalen Akkuspannung betrieben. Thema Strombegrenzung bei den Hoymiles.

der hohe Korrekturfaktor ist unabhängig von der Batteriespannung. Die Leitungswiderstände sind bei allen Batterien die gleichen, egal ob ich am Ende 12, 24, 36 oder 48V anlege. 
Der Strom widerum ist bei höherer Spannung geringer, damit auch die Verluste auf der Leitung, das stimmt.
Meiner Auffassung nach sollte die geringe Kapazität und damit der höhere Innenwiderstand nicht teil des Faktors sein. 

> Reduzierung der Leitungs- und Batterieverluste im Akkubetrieb.

ja

> Die Batterie wird schrittweise, mit sinkender Ladung, sanfter entladen.

ja 

> Der Einfluss des dynamischen Innenwiderstandes der Batterie wird reduziert.

der sollte meiner Meinung nach gar nicht relevant sein. klar, über den Innenwiderstand kann man wieder auf die Gesundheit des Akkus schließen, mit einem BMS gibt es da doch aber in der Regel sowas wie SoH (State of Health) oder die verbleiende Kapazität von der ursprünglichen Gesamtkapazität (bzw. im voll geladenen Zustand 85Ah von ehemals 100Ah Kapazität verbleibend)

> Das könnte auch die Lösung für [Request] DPL: Limit Battery discharge current [...] sein

jain, das würde letzten Endes den Strom ebenfalls begrenzen, aber eben, anders. 
eine allgemeine Begrenzung (bspw. feste 25A für die US2000 nach Datenblatt, oder auch anhand des aktuellen Werts, den das BMS ausgibt), entlädt sich bis auf 0% SoC anders, als dein Vorschlag der schrittweisen Reduzierung, bis die Spannung "stabil" bleibt. eine feste Begrenzung kann konstant bis auf 0% entladen, deinem Vorschlag entnehme ich aber eine kontinuierliche Reduzierung, bis man bei 1% SoC nur noch 0.5A entnehmen kann (beispielhaft)

> Nachteil:
>
>    Im Grenzbereich wenn der Akku bereits im Akku-Limiter-Modus läuft und dann wieder Energie ankommt (Morgens Sonne) könnte die gesamt Energieentnahme kleiner sein als im derzeitigen Voll-Betrieb.

So wie ich es mir vorstelle, wird es auch vorher bereits zu geringerer Entnahme kommen. 


@Manos1966 

> das grossere Problem ist vor allem Vormittags bei wenig Sonne, wenn die Batterie leer ist z.B. 15% und der Victron die Batterie laedt (Spannung ist hoch, ca 53V) aber die Ladeleistung ist kleiner z.B. 500W, als die Maximale Leistung des Wechselrichters z.B. 1500W.
Dann passiert ein "Tisch-Tennis" Effekt : Spannung 53V (laden), Victron liefert nur 400W (es ist bewoelkt), Wechselrichter startet volle Leistung 1500W = Die Spannung geht runter, der Wechselrichter stoppt. Und es faengt wieder von vorne.... das macht die fast leere Batterie noch leerer.

genau das ist ja ein Effekt, der damit erschlagen werden soll.

Die Batterie wird natürlich leerer, da du sie stark entlädst. es kommen 400W rein, es genen 1500W raus (aus dem Gesamtsystem), also entlädst du den Akku kurzzeitig mit 1100W. Wie soll die SPannung da gleich bleiben oder steigen? 😄 
die Spannung ist wahrscheinlich auch nur scheinbar auf 53V, weil der Laderegler lädt, also seine Ladespannung anliegt. Das BMS wird noch keine 53V melden, oder? bzw. wird die Spannung der "leersten" Zelle keine 3.3125V haben. 



theoretisch bräuchte es irgendwie 2 Grenzspannungen, um die Batterie tatsächlich konstant bis auf 0% zu entladen

ich finde keine Diagramme für prismatische Zellen, daher nehme ich einfach mal die hier, um meine Gedanken rüber zu bringen 😄 
https://lygte-info.dk/review/batteries2012/Vapcell%20IFR32700%206500mAh%20G65%20(Blue)%202020%20UK.html

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/36cf9f21-05e8-435f-8e23-8d8e008dcb9b)
Wie man sieht, bricht die Zelle, je nach Last, unterscheidlich stark ein. Das ist uns allen bekannt und auch ok.

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/28ecf89c-1c99-4832-8c5f-b2f5eecfc45e)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/66890545-4681-4ee0-9256-499696a5bae2)

irgendwie bräuchte man 2 untere Spannungen, eine, als absolute Stoppspannung (Lastbefreit, bspw. 2.75V) und eine als untere "Zielspannung" (Lastkorrigiert, bspw. 2V)

Wie man in den oberen Diagrammen sieht, erhohlen sich die Zellen nach 30A und 20A recht schnell. Entladen auf 2V, erhohlt (bzw. spannungsfrei) auf ~2.75V. 



oder man lässt einfach konstant bis auf die 2.75V entladen und nutzt so die restliche Kapazität einfach nicht. Im Falle der 20A Entladelinie würde man ja ~ 0.3Ah nicht nutzen. (so macht es openDTU-oB aktuell, oder? Wenn rein spannungsgeführt und Leitungswiderstandkompensiert geschaltet wird)

Der Vorschlag, der anderswo bereits aufkam, einen allgemeinen Entladestrom für die Batterie festzulegen oder aus dem BMS auszulesen, gefällt mir eigentlich bisher am besten. Ja, eine dynamische Anpassung, wie hier vorgeschlagen, würde die letzten Ah noch herausholen und eine sanfte, dafür längere Entladung am Ende ermöglichen, aber ist das zwingend nötig?


was ich hingegen zukünftig verfolgen würde:

> Zusätzlich habe ich bedenken das der Korrekturfaktor bei einer Anlage statisch ist. Der Widerstand der Kupferleitungen und Sicherungen ändert sich mit der Temperatur

wenn wir die Möglichkeit haben, die Batteriespannung nahe der Batterie (BMS/Shunt) zu messen, und auch die DC Spannung und den fließenden Strom am EIngang kennen, so könnte man den Leitungswiderstand dynamisch berechnen und ggf. eine Warnung im Live View erzeugen, wenn dieser berechnete Wert eine Grenze übersteigt. Es kann ja wirklich mal was an einer Sicherung oder Schraubverbindung sein, was den Widerstand erhöht und am Ende warm/heiß wird. Aber das wäre ein anderes Thema 😄

**Reply by SW-Niko** [reply]
Hallo @Manos1966 und @spcqike ,
vielen Dank für die ausführlichen Rückmeldungen. Ich versuche mal den Akku-Limiter-Betrieb zum besseren Verständnis ein ein Beispiel zu packen. Dazu benötige ich etwas Zeit.... 
@spcqike 
Ja richtig, das ist keine Lösung für [Request] DPL: Limit Battery discharge current [...]. Wenn der maximale Entnahme Strom zum Schutz des Akkus ist.

**Reply by SW-Niko** [reply]
Hallo @spcqike 
Noch eine Anmerkung zu diesem Punkt:
>wenn wir die Möglichkeit haben, die Batteriespannung nahe der Batterie (BMS/Shunt) zu messen, und auch die DC Spannung und den fließenden Strom am EIngang kennen, so könnte man den Leitungswiderstand dynamisch berechnen und ggf. eine Warnung im Live View erzeugen, wenn dieser berechnete Wert eine Grenze übersteigt. Es kann ja wirklich mal was an einer Sicherung oder Schraubverbindung sein, was den Widerstand erhöht und am Ende warm/heiß wird. Aber das wäre ein anderes Thema 😄

Richtig. je näher wir an der Batterie messen desto besser. Auch der Akku-Limiter-Mode profitiert davon. Das kann aber nicht jeder machen und ich suche eine Methode die nahezu jeder Anlage funktioniert und auch für "nicht Elektrotechniker" konfigurierbar ist. 
Darum mag ich den Load-Correction-Factor auch nicht so besonders.

**Comment by SW-Niko** [comment]
Hallo @spcqike und @Manos1966 
Vielleicht wird das Konzept vom Akku-Limiter-Mode bei einer Gegenüberstellung mit dem aktuellen Load-Correction-Factor-Mode in einem Beispiel klarer.

Das wird jetzt etwas technisch, weil ich mich bemüht habe meine Behauptungen mit Daten und Berechnungen zu untermauern.

Als Beispiel nehme ich einen Inverter mit 300W, einem MPPT Laderegler und einen Akku mit 24V / 2KW. Zur einfacheren Berechnung hat er ein lineares Verhalten bezüglich Ladung und Leerlaufspannung. 100% SOC = 27V und 0% SOC = 26V. Daraus ergibt sich folgerndes Schaltbild.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/5110b051-c3f9-4d16-af43-e14a135203b8)

Die Widerstände habe ich von meiner eigenen Anlage übernommen. 

Zum Regeln nehmen wir die Spannung Uinv am Inverter. Der Gesamtwiderstand Rg hat 25mOhm.
Der Akku ist halb voll und hat damit eine Ladung von 1kW / 26,5V


**Aktuelle Methode mit Load-Correction-Factor-Mode:**
Der Widerstand Rg kann in den Load-Correction-Factor umgerechnt werden.
Load-Correction-Factor = Rg/Ubatt = 0.025Ohm/26V = 0,00091.
Wenn der Akku konstant mit 300W entladen wird, dann ist er nach 200 Minuten leer (SOC = 0% / 26V).
Am Inverter messen wir dann eine Spannung von 25,74V aber mit dem richtigen Load-Correction-Factor werden 26V daraus (25,74 + 300 * 0.00091 = 26,01) und wir schalten den Inverter ab.
Halten wir fest: Abschaltung bei 300W und einer Akkuzellenspannung von 26V.

**Neue Methode mit  Akku-Limiter-Mode:**
Anfangs wird der Akku wieder mit 300W entladen. Das machen wir solange bis am Inverter eine Spannung von 26V gemessen wird. Und ab diesem Moment, regeln wir die Leistung runter bis die Schelle von 26V nicht mehr unterschritten wird. Der Akku wird weiter entladen, die Spannung sinkt weiter und die Leistung wird weiter reduziert. Wen 26V erreicht werden schalten wir den Inverter ab.
Wenn die minimale Leistung z.B. bei 30W liegt dann haben wir auf der gesamten Strecke noch einen Spannungsabfall von 31mV.
Halten wir fest: Abschaltung bei 30W und einer Akkuzellenspannung von 26,031V.

**Entladekurven von Load-Correction-Factor-Mode und  Akku-Limiter-Mode**
Das ganze schaut dann so aus. 

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/9e5b0b54-d362-48b5-b4ef-030ea3eebc8e)

X-Achse: Zeit in Minuten
Y-ACHSE: Uakku-LC, Uinv-LC: Werte vom Load-Correction-Factor-Mode
Y-ACHSE: Uakku-AL, Uinv-AL: Werte vom Akku-Limiter-Mode:

Anmerkung: Die wackelige Kurve Uinv-AL kommt von meiner manuellen Anpassung der Entladeleistung. Sollte eigentlich so ab 100 Minuten konstant bei 26V stehen.


**Ergebnis:**
Beide Methoden schalten beim unterschreiten der 26V den Inverter ab.
Die Load-Correction-Factor-Mode ist um 31mV genauer.
Das schaut jetzt erst mal nach einem Punktsieg für die Load-Correction-Factor-Mode aus.

Aber...um das, in der Realität zu schaffen, muss der  Load-Correction-Factor.
1. Für die jeweilige Anlage richtig ermittelt und konfiguriert werden.
2. Er darf sich nicht verändern.

Ich fang jetzt mal aus taktischen Gründen mit Punkt 2 an.
Grob abgeschätzt kann sich der Load-Correction-Factor bei der Beispiel Anlage um ±21% ändern. Diesen Wert habe ich über folgende Eckdaten berechnet.
Bei Kabel, Sicherungen und Verbindungen ist die Temperatur der Hauptfaktor. Ganz grob kann man mit 0.4% pro 1°C rechnen. Wenn man mit ±20C° rechnet, verursacht durch Umgebungstemperatur und Eigenerwärmung dann landet man bei einer Widerstandsänderung von ±8%.
Der Akku-Innenwiderstand hat viele Einflussfaktoren. Ich habe für die Abschätzung folgende Grafik, die ich über Google gefunden habe, als Grundlage genommen.

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/a178c389-7d1f-44a0-a512-f7cf3ced06ba)

Ich bleibe konservativ und lasse den Akku im Keller stehen bei einer Zell Temperatur zwischen 10°C bis 30°C und beschränke mich auf 100% SOC bis 20% SOC.
Wir bekommen dann bei einem  Akku-Innenwiderstand von 5.2mOhm – 2.5mOhm! Oder 3,85Ohm ±35%.

Wenn ich diese Daten auf meine Beispielanlage umrechne dann komme auf Rg = 25mOhm ± 21%.
Die Toleranz von Rg ist hier kleiner, weil in dieser Anlage der hohe Leitungswiderstand die starke Toleranz des Akkus abmildert.
Noch deutlicher wird die Problematik, wenn ich die ±21% jetzt auf die Abschaltspannung bei 300W Entnahmeleistung umrechne. Wir landen dann bei 26V ± 60mV.

Und nun zu Punkt 1. Die richtige Ermittlung des  Load-Correction-Factors. Wer macht das richtig und ermittelt den  Load-Correction-Factor bei:
- Mittlerer Temperatur
- Mittlerer Entladung
- Mittlerem Ladezustand
Wer??? 
Wenn man am oberen oder unteren Ende liegt, und nicht schön in der Mitte, dann kann die berechnete Akku-Zellenspannung bei einem 24V Akku um 120mV daneben liegen. Das kann dazu führen das die BMS eingreift oder das man Energie im Akku nicht nutzt.

Ich habe bewusst ein einfaches Beispiel gewählt. In Realität gibt es noch weitere Faktoren die den Load-Correction-Factor beeinflussen:
- Alterung
- Anlagen mit mehreren Akkus oder Invertern
- Energie die vom MPPT eingespeist wird
- ….

Puuu...Ich hoffe ihr habt durchgehalten. Noch Fragen bis hierher?
Im nächsten Teil vergleiche ich beide Methoden bei dem Zustand: „Akku fast leer und die Sonne beginnt wieder Energie zu liefern“.
@Manos1966  Das könnte deinen "Tisch-Tennis" Effekt erklären.

**Comment by spcqike** [comment]
Hallo,

danke für die ausführliche Darstellung.

Generell finde ich den Vorschlag gut. Zumal er, sollte es umgesetzt werden, wahrscheinlich eine Möglichkeit und keine Pflicht darstellen wird, oder?

Ich würde, wie gesagt, den Innenwiderstand der Zellen nie mit in den Korrekturfaktor einfließen lassen. Ich möchte lediglich die Leitungs- und Verbindungsverluste korrigieren (so, wie es eine zusätzliche Sense-Leitung an einem Laderegler auch macht). Wenn die Zelle unter Last zu stark einbricht, ist das ein Zeichen von Alterung oder sonstigen Defekten (oder auch einfache Überlast), die ich nicht per Software kompensieren würde. Wie du richtig sagst, den Gesamt-Korrekturfaktor (inkl. Zelle) kann wahrscheinlich niemand zu Hause sinnvoll ermitteln. Dann wird er empirisch ermittelt und so lange hoch genommen, bis die Werte "passen". ein zu hoher Faktor führt aber dazu, dass die Zelle "immer voll" oder "immer gut" ist. 
Den reinen Leitungswiderstand kann man hingegen sehr gut ermitteln. Einfach der Spannungsabfall von der Batterie bis zum Wechselrichter unter Berücksichtigung des fließenden Stroms. R=U/I. Oder, empirisch den Wert in der openDTU so lange erhöhen, bis die unter Last gemessene DC-Spannung am Eingang identisch zur mit einem Multimeter gemessenen Spannung an der Batterie ist.


Ein weiterer Vorteil des vorgeschlagenen AL Modus: Durch geringere Ströme hat man automatisch geringere Verluste (P=RI²) 
ein Nachteil: bei geringerer Auslastung sind die Wechselrichter weniger effizient (oder?) 

Du zeigst es ja schön:
natürlich nimmt die Entladeleistung ab. dafür erhöht sich die Zeit von 200 Minuten auf über 500 Minuten. Im Mittel entlädt man in deinem Beispiel also mit ~120W, anfangs mehr, am Ende nur noch 30W bis es stoppt. 

Ist diese Regelung immer sinnvoll? Ich stelle mir grade vor, eine 5kWh Batterie hat früh um 07:00 nur noch 500Wh, 08:00 geht die Sonne auf, aber aufgrund der tiefen Entladung speist man nur noch 50W ein, kauft bspw. 150W aus dem Netz ein. ab 08:00 Uhr lädt die Batterie wieder auf, sie war nicht leer, selbst "geernteter Strom" wurde nicht genutzt, stattdessen wurde aus dem Netz bezogen.

Ich glaube, ich würde eine Regelung, wie von dir vorgeschlagen, durchaus testen und gucken, wie es läuft. In meinem Kopf wäre eine "einfachere" Regelung, so wie bisher, ggf. erweitert um einen maximalen Entladestrom aus der Batterie, aber dennoch mein Favorit 😄 

> Das könnte deinen "Tisch-Tennis" Effekt erklären.

der Tisch-Tennis-Effekt kann auch erklärt werden, wenn die Korrektur falsch ist, die Entladung zu hoch (starker Einbruch der Batterie) oder die Grenzspannungen zu nah bei einander liegen.

**Comment by SW-Niko** [comment]
Im Teil 2 schau ich mir das Verhalten der beiden Methoden an, wenn die Batterie fast leer ist und die Sonne wieder Energie liefert.

Ich bleibe bei der Anlage von Teil 1.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/60b0cb60-0fb4-4acb-8d78-2209f869bd8d)

Aber diesmal ist der Akku fast leer. 10% SOC, 26,05V.

**Aktuelle Methode mit Load-Correction-Factor-Mode:**
Load-Correction-Factor = Rg/Ubatt = 0.025Ohm/26V = 0,000962.
Der Akku wird konstant mit 300W entladen. Jetzt werden zusätzlich 200W über den MPPT eingespeist.
Ich erkläre das Verhalten mal mit dem Ohmschen Gesetz: „Spannung = Strom * Widerstand“.
Wenn über den MPPT nichts eingespeist wird (Imppt = 0), dann ist Iakku = Inv und der Spannungsabfall über die Widerstände ist Iakku * R-Akku + Iakku * R-Inv2 + Iakku * R-Inv1.
Kommt jetzt Imppt dazu, dann fließt über R-Inv1 immer noch der gleiche Strom aber über R-Akku und R-Inv2 um Imppt weniger Strom. Weniger Strom verursacht weniger Spannungsabfall und die Spannung Uinv steigt ungefähr um diesen Anteil an.
Uinv_neu = Uinv_alt + (Iakku – Imppt ) * (R-Akku + R-Inv2). Der Load-Correction-Factor-Mode rechnet aber immer noch so als ob der Iinv durch alle 3 Widerstände fließen wurde und damit wird die Spannung Uzellen zu hoch berechnet.

Diese einfache Berechnung ist nicht ganz korrekt, liefert aber ein Ergebnis mit vernachlässigbarem Fehler.

Hier noch mal eine Tabelle in der die MPPT Leistung von 0W – 400W geändert wird.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/9e06e10e-0b77-4cc5-b892-d5a9e3d6fa2b)

Man beachte den Unterschied zwischen der Zeile „Akku Zellen Voltage“ und der Zeile „Akku Zellen Correction“ was der berechneten Zellenspannung über die Formel des Load-Correction-Factor-Mode entspricht. Bei 200W MPPT Leistung liegt die Berechnung in diesem Beispiel um 0,131V zu hoch.
Kritisch ist der Bereich von MPPT 1-299W, weil der Akku immer noch entladen wird.

Halten wir fest: Der zusätzlich eingespeiste Strom über den MPPT verfälscht die Berechnung der Akkuspannung und liefert immer eine höhere Spannung als sie in Realität ist. Es besteht die Gefahr das über die DTU zu spät abgeschaltet wird oder das das BMS eingreift.



**Neue Methode mit  Akku-Limiter-Mode:**
Wie wir im Teil 1 gesehen haben, hat der Akku-Limiter-Mode, die Leistung vor dem Abschalten auf 30W reduziert. Jetzt wird, wie oben, zusätzlich 200W über den MPPT eingespeist. Die Regelung des Akku-Limiter-Mode erhöht die Leistung am Inverter auf ca. 200W.
Mit anderen Worten, die Regelung folgt in diesem Fall automatisch dem MPPT.
Steigt die MPPT Leistung weiter an und kommt über 300W, dann wird der Akku geladen und der Akku Limiter-Mode verlassen.

**Ergebnis:**
Der Load-Correction-Factor-Mode kann nicht den gesamten Spannungsabfall kompensiert. Die Berechnung stimmt nur bis zum Knotenpunkt an dem der MPPT einspeist.
Der Faktor muss zur Fehlervermeidung auf Load-Correction-Factor = Rg/Ubatt = 0.008Ohm/26V = 0,0003 reduziert werden.
Der „Akku-Limiter-Mode“ macht diesen Fehler nicht.

**Comment by Feldsalat** [comment]
Ich finde die Idee gut, würde auch mein Problem mit der Grundlast lösen, das ich durch meine zusätzliche USV-Nutzung der Batterie habe.
